01_oop_and_ood/src/Shape.ts

```diff
@@ -0,0 +1,63 @@
+import { Point } from "./Point";
+
+export abstract class Shape {
+    protected color: string;
+    protected filled: boolean;
+    points: Point[];
+
+    constructor(points: Point[])
+    constructor(points: Point[], color: string, filled: boolean)
+    constructor(...args: any[]) {
```

Comment:
Same here, you’ve defined that you can pass an array of points, or array of points, color and filled, but you are waiting for any kind of arguments and any count of them. Please, add
`constructor(points: Point[], color = 'green', filled = 'true')` instead `constructor(...args: any[])`

that will help to rid out checking of types `typeof args[1] === 'string' && typeof args[2] === 'boolean'` - typescript is doing it for you, also it will help you to rid out of checking whether we have only points or we have filled and color.

Reply:
done